### PR TITLE
Code samples

### PR DESCRIPTION
**Pull request by cionescu**

### CODE REVIEW

File: src/app/link/code.service.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export class CodeService {
+  private apiUrl = `${ENV.apiUrl}/code_snippet`;
+
+  constructor(private http: Http) { }
+
+  all(boardType: string, linkTypes: string[]): Promise<Code[]> {
+    const search: URLSearchParams = new URLSearchParams();
+    search.set('board_type', boardType);
+    for (const type of linkTypes) {
+      search.append('link_types[]', type);
```
Comment (cionescu) [inline src/app/link/code.service.ts:18]: This is how you can send an array as a query param

File: src/app/link/code.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export interface CodeInterface {
+  code: string;
+  name: string;
+}
+
+export class Code {
+  private code: string;
+  private name: string;
+
+  constructor(codeInterface: any) {
```
Comment (paul110) [inline src/app/link/code.ts]: aici, de ce nu ii dai de tipul :codeInterface ?

File: src/app/link/code.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+    return this.http
+      .get(`${this.apiUrl}.json`, {search})
+      .toPromise()
+      .then( (response) => Array.from(response.json(), (code: CodeInterface) => new Code(code) ))
```
Comment (paul110) [inline src/app/link/code.service.ts]: hai sa folosim .map deci :)), ca am schimbat acolo unde ai zis tu la mine in cod

File: src/app/link/code.service.ts (hunk 1 above)
Comment (paul110) [reply src/app/link/code.service.ts:18]: nice

**Pull request merged**